doc/plutus/explanations/plutus-tools-component-descriptions.rst

```diff
@@ -0,0 +1,241 @@
+| Plutus contract model testing                                                |
++==============+===============================================================+
+| Location     | ``plutus-apps/plutus-contract``                               |
++--------------+---------------------------------------------------------------+
+
+`Plutus contract testing model <https://github.com/input-output-hk/plutus-apps/blob/main/plutus-contract/src/Plutus/Contract/Test/ContractModel.hs>`_ is used for testing prototype Plutus contracts with *contract models*, using the framework provided by :hsmod:`Plutus.Contract.Test.ContractModel`. 
+This framework generates and runs tests on the Plutus emulator, where each test may involve a number of emulated wallets, each running a collection of Plutus contracts, all submitting transactions to an emulated blockchain. 
+Once you have defined a suitable model, then QuickCheck can generate and run many thousands of scenarios, taking the application through a wide variety of states, and checking that it behaves correctly in each one. 
+
+   See also: 
+   
+   * :doc:`Property-based testing of Plutus contracts <../tutorials/contract-testing>` 
+   * :doc:`Testing Plutus Contracts with Contract Models <../tutorials/contract-models>` 
+
+3. Contract monad emulator
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
++--------------+---------------------------------------------------------------+
+| Contract monad emulator                                                      |
++==============+===============================================================+
+| Location     | ``plutus-apps/plutus-contract``                               |
++--------------+---------------------------------------------------------------+
+
+`Contract monad emulator <https://github.com/input-output-hk/plutus-apps/tree/main/plutus-contract>`_ is a library that provides an environment for emulating the blockchain. 
+The environment provides a way for writing traces for the contract which are sequences of actions by simulated wallets that use the contract. 
+The component is highly dependent on the Contract API (Contract monad). 
+
+4. Contract API (also known as Contract monad)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
++--------------+---------------------------------------------------------------+
+| Contract API                                                                 |
++==============+===============================================================+
+| Location     | ``plutus-apps/plutus-contract``                               |
++--------------+---------------------------------------------------------------+
+
+`Contract API <https://github.com/input-output-hk/plutus-apps/tree/main/plutus-contract>`_ is a logical component within the Plutus Contract package, providing an effective system for describing smart contracts that interact with wallets, DApps and the blockchain. 
```

Comment:
effective -> effect?

Reply:
interact with wallets, DApps and the blockchain -> interact with wallets, dApps, a chain indexer and the blockchain

Reply:
I've incorporated your comments. By the way, our IOG style guide says that we're supposed to say "DApps" rather than "dApps."

Reply:
Nice to know :)